shoes-core/spec/shoes/console_spec.rb

```diff
@@ -0,0 +1,215 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe Shoes::Console do
+  # include_context 'dsl app'
+
+  def call_button_mock_block(button_content)
+    button_content.gui.instance_variable_get(:@mock_clickable_block).call
+  end
+
+  let(:console) { Shoes.console }
+
+  let(:sample_message_array) { [[:debug, 'debug message'], [:warn, 'warn message'], [:error, 'error message']] }
+  let(:formatted_message_output){ "Debug\n debug message\n\nWarn\n warn message\n\nError\n error message\n\n" }
+
+  let(:mock_dialog) { Shoes::Mock::Dialog.new.dialog_chooser }
+
+  before(:each) do
+    # Depending on spec order, these can be leftover and contaminate other specs
+    console.instance_variable_set(:@app, nil)
+    console.instance_variable_set(:@messages, [])
+  end
+
+  describe '#initialize' do
+    it 'must reset @messages as []' do
+      console.instance_variable_set(:@messages, ['placeholder'])
```

Comment:
You shouldn't need to set the instance variable here directly. With the reader already on `messages`, a simple `console.messages << 'placeholder'` (assuming our startup is clean) would suffice.